### PR TITLE
Enhance the functionality of the `bump_version.sh` script

### PR DESCRIPTION
**Pull request by mcdonnnj**
#  #

## 🗣 Description ##

This pull request adds some functionality to the `bump_version.sh` script and DRYs out some of the logic.

> [!Note]
> If/when this pull request is approved I plan on porting the changes over to other skeletons that have a `bump_version.sh` script.

## 💭 Motivation and context ##

I have used custom tokens for prerelease versions in some of my projects (`dev` instead of `rc` as an example) to better articulate what the version represents. I thought about it and determined that building the functionality into the script would be useful. While I was making the changes to support custom tokens I realized it would also make sense to do some general cleanup of the script.

## 🧪 Testing ##

Automated tests pass. I verified that the script works as expected both with and without custom tokens for `build`/`prerelease` actions. I also verified that other actions worked as expected.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All new and existing tests pass.

### CODE REVIEW

File: bump_version.sh
Comment (dav3r) [inline bump_version.sh:61]: If the provided token doesn't match the existing token, it looks `semver` doesn't modify the version or throw an error:
```
>>> import semver
>>> version_info = semver.VersionInfo.parse("1.2.3-rc.4")
>>> version_info.bump_prerelease()
Version(major=1, minor=2, patch=3, prerelease='rc.5', build=None)
>>> version_info.bump_prerelease(token="foo")
Version(major=1, minor=2, patch=3, prerelease='rc.5', build=None)
```

Is that the desired behavior here?  Can you please share output of a manual run of this code in this case so that we can see how the script behaves?

**Review by dav3r — COMMENTED** [review]
This looks like a nice improvement, but I want to get clarification on one thing before I approve.

File: bump_version.sh
Comment (mcdonnnj) [reply bump_version.sh:61]: You are not mirroring how this is done in the script in your example. The note in the help section clarifies that the provided token is not used if it does not match, but it will still version bump:

```console
>>> import semver
>>> old_version = "1.2.3-rc.4"
>>> semver.bump_prerelease(old_version)
'1.2.3-rc.5'
>>> semver.bump_prerelease(old_version, token="test")
'1.2.3-rc.5'
>>> old_version = "1.2.3"
>>> semver.bump_prerelease(old_version)
'1.2.3-rc.1'
>>> semver.bump_prerelease(old_version, token="test")
'1.2.3-test.1'
```

Edit:
After looking at your example it is actually doing things correctly. The `bump_prerelease()` method returns a _new_ `Version` object:

```console
>>> import semver
>>> version_info = semver.VersionInfo.parse("1.2.3-rc.4")
>>> version_info.bump_prerelease()
Version(major=1, minor=2, patch=3, prerelease='rc.5', build=None)
>>> version_info.bump_prerelease()
Version(major=1, minor=2, patch=3, prerelease='rc.5', build=None)
```

So it is still incrementing the version but using the old token. This is the expected, and documented, behavior.

Comment (dav3r) [reply bump_version.sh:61]: Alright, I'm tracking now.  I was a little confused before, but this all makes sense.

I don't totally love the silent no-op (not exactly the right phrase, since it does bump the version, but it ignores the token) if you attempt to use a different token than the current version, but your note in the usage text does indeed mention this behavior.

**Review by dav3r — APPROVED** [review]
👍 👍

File: bump_version.sh
Comment (mcdonnnj) [reply bump_version.sh:61]: Yes I'm not a fan of it silently ignoring the provided token, but the Python logic completely ignores the token if the version string is already a prerelease per the [sourec code](https://github.com/python-semver/python-semver/blob/8fe4ef678e1cbacd904fbd9f67a8cb3f971642b0/src/semver/version.py#L326-L333). Making sure to explicitly mention/document the behavior was the only thing I could see doing since I'd rather expose the functionality even with the caveat.

File: bump_version.sh
Comment (jsf9k) [inline bump_version.sh]: I have been thinking for a while that this `git push` should be left to the user when he or she is ready.  Removing it will cut down on needless GitHub Actions runs, which is better for the environment.
```suggestion
```

**Review by jsf9k — APPROVED** [review]
I made one suggestion, which need not impede the approval of this pull request.

File: bump_version.sh
Comment (mcdonnnj) [reply bump_version.sh]: I am completely on board with that (and it has been a peeve of mine for a while). @dav3r any concerns?

Comment (dav3r) [reply bump_version.sh]: None- I wholeheartedly agree.

**Review by jasonodoom — APPROVED** [review]

**Pull request merged**
